landscaper/common/generate/blueprint/generate.go

```diff
@@ -135,13 +181,78 @@ func cleanDirectory(directory string) error {
 // renderBlueprint creates the blueprint.yaml in the blueprint directory based on the `blueprintTemplate`
 // The blueprint resource must contain fields defining its import fields. Only using a `ref` to a JSONSchema file is not possible.
 // Hence, this function takes the properties of the root schema and renders them as top-level import fields to the blueprint.
-func renderBlueprint(definition common.OpenAPIDefinition, blueprintTemplate *template.Template, blueprintDirectory string) error {
-	var topLevelFields []BlueprintImport
-	requiredFields := sets.NewString(definition.Schema.SchemaProps.Required...)
+func renderBlueprint(importDefinition common.OpenAPIDefinition, exportDefinition *common.OpenAPIDefinition, blueprintTemplate *template.Template, blueprintDirectory string, exportExecutionFileName *string, containerDeployerName *string) error {
+	importsFields, _, err := getTopLevelFields(importDefinition)
+	if err != nil {
+		return err
+	}
+
+	data := map[string]string{
+		"imports": string(importsFields),
+	}
+
+	if exportDefinition != nil {
+		exportsFields, exportedFields, err := getTopLevelFields(*exportDefinition)
+		if err != nil {
+			return err
+		}
+		data["exports"] = string(exportsFields)
+
+		err = writeExportExecutions(exportedFields, blueprintDirectory, exportExecutionFileName, containerDeployerName)
+		if err != nil {
+			return err
+		}
+	}
+
+	var ccdScript bytes.Buffer
+	if err := blueprintTemplate.Execute(&ccdScript, data); err != nil {
+		return err
+	}
+
+	return ioutil.WriteFile(fmt.Sprintf("%s/blueprint.yaml", blueprintDirectory), ccdScript.Bytes(), 0640)
```

Comment:
The `io/ioutil` pkg was deprecated in go1.16 - ref https://go.dev/doc/go1.16#ioutil. In https://github.com/gardener/gardener/issues/4020 we also spend effort to clean up all `io/ioutil` usage, hence we don't want to introduce new ones.

```suggestion
	return os.WriteFile(fmt.Sprintf("%s/blueprint.yaml", blueprintDirectory), ccdScript.Bytes(), 0640)
```